File: packages/contracts/src/enums.ts
// packages/contracts/src/enums.ts

export enum PostType {
  EXPRESSION = 'expression',
  CLAIM = 'claim',
  INVESTIGATION = 'investigation',
}

export enum EvidenceDirection {
  SUPPORTS = 'supports',
  CONTRADICTS = 'contradicts',
}

export enum ConfidenceState {
  NO_REVIEW = 'no_review',
  COMMUNITY_REVIEWED = 'community_reviewed',
  CONFLICTING_EVIDENCE = 'conflicting_evidence',
}

export enum ContentStatus {
  ACTIVE = 'active',
  LOCKED = 'locked',
  REMOVED_ILLEGAL = 'removed_illegal',
}

export enum ModerationActionType {
  REMOVED_ILLEGAL = 'removed_illegal',
  LOCKED = 'locked',
}

export enum UserStatus {
  ACTIVE = 'active',
  SUSPENDED = 'suspended',
}

File: apps/api/app/core/enums.py
# apps/api/app/core/enums.py

from enum import Enum

class PostType(str, Enum):
    expression = "expression"
    claim = "claim"
    investigation = "investigation"

class EvidenceDirection(str, Enum):
    supports = "supports"
    contradicts = "contradicts"

class ContentStatus(str, Enum):
    active = "active"
    locked = "locked"
    removed_illegal = "removed_illegal"


class UserStatus(str, Enum):
    ACTIVE = "active"
    SUSPENDED = "suspended"

class FeedMode(str, Enum):
    latest = "latest"
    trending = "trending"


class SecurityEventType(str, Enum):
    login_success = "login_success"
    login_failed = "login_failed"
    account_locked = "account_locked"
    password_reset_requested = "password_reset_requested"
    password_reset_completed = "password_reset_completed"
    refresh_reuse_detected = "refresh_reuse_detected"
    session_revoked = "session_revoked"
    logout = "logout"
    refresh_token_reuse = "refresh_token_reuse"
    resend_verification = "resend_verification"
    suspicious_login_country = "suspicious_login_country"
    impossible_travel_login = "impossible_travel_login"
    suspicious_login_ip = "suspicious_login_ip"
    suspicious_login_device = "suspicious_login_device"
